src/utils/AppIconWrapper.ts

```diff
@@ -0,0 +1,68 @@
+import { Alert, Platform } from 'react-native';
+import { changeIcon, getIcon as getCurrentIcon, resetIcon } from 'react-native-change-icon';
+
+const isAndroid = Platform.OS == 'android';
+const ANDROID_ICONS = {
+  default: 'io.hexawallet.keeper.MainActivityDefault',
+  keeperPrivate: 'io.hexawallet.keeper.MainActivityKp',
+};
+const iOsKeeperPrivateIcon = 'KpAppIcon';
+
+export const AppIconWrapper = () => {
+  const getIcon = async () => {
+    const response = await getCurrentIcon();
+    return response;
+  };
+
+  const changeToDefaultIcon = async () => {
+    if (isAndroid) {
+      const currentIcon = await getCurrentIcon();
+      if (currentIcon != ANDROID_ICONS.default) {
+        Alert.alert(
+          'Keeper Private',
+          'You have been downgraded from Keeper Private, the app will now restart to apply the changes.',
```

Comment:
We have failed to verify your Keeper Private subscription or it has expired. The app will now restart to apply the changes.